src/js/angular/security/controllers.js

```diff
@@ -484,10 +484,29 @@ securityCtrl.controller('CommonUserCtrl', ['$rootScope', '$scope', '$http', 'toa
             return $scope.user && !$scope.user.appSettings.DEFAULT_INFERENCE;
         };
 
+        $scope.showTagError = false;
+
         $scope.addCustomRole = function (role) {
+            $scope.showTagError = false;
             role.text = role.text.toUpperCase();
             return role;
         };
+
+        $scope.validateTag = function (tag) {
+            if (tag.text.length < 2 ) {
+                $scope.showTagError = true;
+                return false;
+            }
+            $scope.showTagError = false;
+            return true;
+        };
+
+        $scope.checkForBackspace = function(event) {
+            // If the key pressed is the backspace or delete key, the tag error message will be hidden
```

Comment:
There is another case where the error message remains visible and it is when I select the character and cut it either via [ctrl+x] or via the context menu. This would be very rare case though and if you find it hard to implement, you can just drop a comment here so that it's documented as a known issue at least.